crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -392,6 +392,30 @@ impl<'a> Resolver<'a> {
         }
 
         let span = path.span();
+        if let Some(type_alias_type) = self.lookup_type_alias(path.clone()) {
+            let mut args = vecmap(args, |arg| self.resolve_type_inner(arg, new_variables));
+            let expected_generic_count = type_alias_type.borrow().generics.len();
+
+            if args.len() != expected_generic_count {
+                self.push_err(ResolverError::IncorrectGenericCount {
+                    span,
+                    struct_type: type_alias_type.borrow().to_string(),
+                    actual: args.len(),
+                    expected: expected_generic_count,
+                });
+
+                // Fix the generic count so we can continue typechecking
+                args.resize_with(expected_generic_count, || Type::Error);
+            }
+
+            // Type Alias is monomorphized here because we do type checking before monomorphization
+            // If this is done in normal monomorphization phase type checking will return errors
+            // TODO(ethan): not sure if this is the best practice though
+            let typ = type_alias_type.borrow().get_type(&args);
+
+            return typ;
+        }
+
```

Comment:
This looks fine to me (sans one item, see below). The comment can be removed since its using the term "monomorphize" incorrectly but other than that the actual code looks fine. Not adding aliases directly to the type representation that the type checker uses is a tradeoff. It means we don't have to handle them in the type checker which makes things easier, but also means users won't see type aliases in their error messages which can make some types appear larger than necessary. Although other times we'd need to expand out the aliases anyway in case the type error is within them so it isn't too major a point.

The one item that may be missing is type aliases referring to other aliases, e.g:
```rs
type Three = Two<u8>;
type Two<A> = One<A, u32>;
type One<A, B> = (A, B);
```
Can you confirm that this works?